### DISCUSSION TITLE
Way to "pause" BLE scanning in shell integration

### DISCUSSION THREAD

**Discussion by lesensei** [mKeRix/room-assistant#430] [Ideas]
Hello,

I am happily using room-assistant to scan for the presence of BLE smartbands and to receive advertisements from some xiaomi devices. This works great and I want to thank everyone and particularly @mKeRix for it :-)

I have another need that I believe is pretty rare: the people in my household that have the bands with them don't necessarily need or want the companion app on their phones (or have a phone, for that matter). So we might not get alerts when the bands batteries are low and the presence detection will end up failing :-/

So here's my idea: the bands (Mi band 1s and Mi band 3) unfortunately do not advertise their battery level, but it is possible to get it by connecting to the devices with `gatttool`. The only thing stopping me from scripting this is that the bluetooth interface is actually in use by the BLE integration. I have looked at the Bluetooth Classic integration code and I think I grasped that it "pauses" the BLE scan while it does its own thing. Am I getting this right ?

I guess the right way to do what I want would be to build a new integration, but I don't think I'll manage to do that with my complete lack of java/typescript knowledge. That leaves me to try it with the shell integration, and with the following question: would it be possible to add an option to the shell integration so that it pauses the BLE process while it runs ?

Thanks a lot for your work !

**Comment by lesensei** [comment]
Actually, I've been looking at hooking a bluetooth speaker to my system. This would also require bluetooth scanning to be paused, as I understand it, and for this reason I guess the best solution would be to duplicate the "inquiries switch" functionality of the bluetooth classic into the BLE one. Would that be something you would consider ?
(I guess as last resort I could just stop the room-assistant service with something like mqtt-launcher, so this feature is not entirely critical in the end)

**Reply by mKeRix** [reply]
I haven't done this so far because BLE is quite passive and doesn't interfere much, but your use case does make sense. I think it would be a decent addition to room-assistant - could you open this as a feature request issue so that I can track it?

As for the battery levels in your original post - there is a feature request open for doing this straight from room-assistant (#160). I've laid the groundwork for this while working on the iPhone BLE topic, which should make that somewhat easy to implement. I just want to finish what I'm currently working on first. :)

**Reply by lesensei** [reply]
Well thanks for even considering it ! I opened https://github.com/mKeRix/room-assistant/issues/458 for this.

As for https://github.com/mKeRix/room-assistant/issues/160 I'm not sure I understand what's gonna happen there, since the way I understand it, it's been solved with the batteryMask stuff and relies on advertisements made by device. Did I miss something ?

**Reply by mKeRix** [reply]
#160 also covers an approach for BLE devices that don't expose their battery information as part of the ID - which is why it is still open. Plan is to use the battery service that a lot of BLE devices expose to gather the needed information.

**Reply by lesensei** [reply]
Ah, yeah, sorry, I don't know how I missed it :-/
I believe mibands don't use the battery service, though, but maybe we'll be able to hook into the code for specific devices that have different ways of reporting battery information (mibands from 3 onwards, however, need specific authentication to do that, which might compromise that capability).
Thanks again !
